sdks/go/pkg/beam/core/runtime/exec/pardo_test.go

```diff
@@ -132,10 +132,12 @@ func BenchmarkParDo_EmitSumFn(b *testing.B) {
 	}
 	go func() {
 		if err := p.Execute(context.Background(), "1", DataContext{}); err != nil {
-			b.Fatalf("execute failed: %v", err)
+			b.Errorf("execute failed: %v", err)
```

Comment:
Well looks like the issue with the blocking was because I was checking before close(process) was called. I like the channel method better because it's a little more clear how the error is handled compared to the error + return.